### PR TITLE
Make sure Flax pipelines can be loaded into PyTorch

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

The test failure is due to an edge case of this repo: https://huggingface.co/hf-internal-testing/tiny-stable-diffusion-lms-pipe/blob/7a9063578b325779f0f1967874a6771caa973cad/model_index.json#L2

As you can see the repo contains both Flax and PyTorch weights, but Flax weights are exceptionally the principal weights here. Therefore the `model_index.json` has `FlaxStableDiffusionPipeline` as the main class. 

To allow loading such a pipeline in PyTorch we need to make sure that we never try to import a Flax pipeline class.

### CODE REVIEW

File: src/diffusers/pipelines/pipeline_utils.py [hunk 1]
```diff
@@ -1438,6 +1443,9 @@ def download(cls, pretrained_model_name, **kwargs) -> Union[str, os.PathLike]:
             ]
 
             # retrieve passed components that should not be downloaded
+            import ipdb
+
+            ipdb.set_trace()
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/pipeline_utils.py:1448]:
```suggestion
```

**Comment by patrickvonplaten** [comment]
@DN6 @sayakpaul @yiyixuxu can you have a look here just to understand why the fast tests were failing?
